### DISCUSSION TITLE
Improve workflow and/or documentation for using a git version of this core

### DISCUSSION THREAD

**Discussion by matthijskooijman** [stm32duino/Arduino_Core_STM32#1335] [Ideas]
**Is your feature request/improvement related to a problem? Please describe.**
Currently, using a git master version of this core is a bit tricky, because it's non-trivial to install the right tools (gcc/CMSIS).

There is some documentation about this [on the wiki](https://github.com/stm32duino/wiki/wiki/Using-git-repository), which explains how to install the core through the board manager to get the tools, remove the installed files (leaving the tools) and replacing it with a git clone. However, when the git version needs newer tool versions that the latest released version (as now), this does not work, and the documentation only says "you must install the correct versions of the required dependencies", without any guidance as to how (I presume this involves manually downloading and extracting zipfiles, which isn't pretty).

Now, I noticed that the https://github.com/stm32duino/BoardManagerFiles repo actually has a `dev` branch, which offers the newer versions of these tools (as required by git master) and a `2.0.0-dev` version of the core which depends on them. So if you use the JSON file from this dev branch, you can let the Arduino IDE install the right tools (and then apply the same trick with deleting and replacing the core as documented). So maybe it would be good to document this dev branch approach on the wiki?

However, this approach still has some downsides:
 - Install 2.0.0-dev actually installs the 1.9.0 released core version, not an actual git version. It's not entirely trivial to install the latest version automatically, though (though you could use a github action to update the json file on *every* push to master maybe).
 - You still cannot have the released 1.9.0 version and a dev version installed together.

So I was wondering about another approach: Could the 2.0.0-dev version maybe be a *dummy* core (that just installs an empty zip and no boards), that serves to just install the dependencies? If it can use a different name (or different architecture), so it can be installed alongside an actual 1.9.0 release, that might be even better.

I haven't tried this approach, but wanted to share the idea for consideration.

**Comment by fpistm** [comment]
@matthijskooijman 
In fact I've planned to use the dev branch to provide the package of the master like an nightly or something like that using a GA.
I saw your mail in the Arduino developers mailing list about: `[Developers] Autogenerating package_index.json files (e.g. from Github Actions?)`

This is something I've in mind since a while.
